js/CobrowseIO.js

```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
 
 const emitter = new NativeEventEmitter(CobrowseIONative)
 
+class Session {
+  constructor (session) {
+    this._session = session || {}
+    this._listen()
+  }
+
+  _update (session, onEnd) {
+    if (this._session.id === undefined || (session && this._session.id === session.id)) {
+      this._session = session
+      if (typeof onEnd === 'function') onEnd()
+    }
+  }
+
+  _listen () {
+    const requests = emitter.addListener(CobrowseIONative.SESSION_REQUESTED, (session) => this._update(session))
+    const updates = emitter.addListener(CobrowseIONative.SESSION_UPDATED, (session) => this._update(session))
+    const ended = emitter.addListener(CobrowseIONative.SESSION_ENDED, (session) => {
+      this._update(session, () => {
+        requests.remove()
+        updates.remove()
+        ended.remove()
+      })
+    })
+  }
+
+  addListener (eventType, listener) {
+    return emitter.addListener(eventType, (session) => {
+      this._update(session, () => {
+        listener(this)
+      })
+    })
+  }
+
+  get id () {
+    return this._session.id
+  }
+
+  get code () {
+    return this._session.code
+  }
+
+  get state () {
+    return this._session.state
+  }
+
+  get full_device () {
+    return this._session.full_device
+  }
+
+  get remote_control () {
+    return this._session.remote_control
+  }
+
+  get agent () {
+    return this._session.agent
+  }
+
+  activate () {
+    return CobrowseIONative.activateSession().then((session) => {
+      this._session = session
+    })
+  }
+
+  end () {
+    return CobrowseIONative.endSession()
+  }
+
+  hasAgent () {
+    return !!(this._session.agent)
+  }
```

Comment:
I don't think we have this on the agent API? I know it's on the Android API but probably not actually needed as we have the agent getter

Reply:
It's on the web sdk and the android sdk. I felt it was appropriate to also expose on the react-native sdk for feature-parity.

Reply:
Oh, actually, I misinterpreted the web sdk. What I thought of as "hasAgent" is just a method called "agent" which returns the agent object or the value `false`.